bazel/repositories.bzl

```diff
@@ -0,0 +1,146 @@
+    urls = ["https://github.com/envoyproxy/envoy/archive/v1.17.0.tar.gz"],
+    strip_prefix = "envoy-1.17.0",
+  )
+
+def com_github_grpc_grpc():
+  http_archive(
+    name = "com_github_grpc_grpc",
+    sha256 = "7372a881122cd85a7224435a1d58bc5e11c88d4fb98a64b83f36f3d1c2f16d39",
+    urls = ["https://github.com/grpc/grpc/archive/v1.34.0.tar.gz"],
+    strip_prefix = "grpc-1.34.0",
+  )
+
+def com_google_googletest():
+  http_archive(
+    name = "com_google_googletest",
+    sha256 = "9dc9157a9a1551ec7a7e43daea9a694a0bb5fb8bec81235d8a1e6ef64c716dcb",
+    strip_prefix = "googletest-release-1.10.0",
+    urls = ["https://github.com/google/googletest/archive/release-1.10.0.tar.gz"],
+  )
+
+def com_google_abseil():
+  http_archive(
+    name = "com_github_abseil-cpp",
+    sha256 = "e3812f256dd7347a33bf9d93a950cf356c61c0596842ff07d8154cd415145d83",
+    strip_prefix = "abseil-cpp-5d8fc9192245f0ea67094af57399d7931d6bd53f",
+    urls = ["https://github.com/abseil/abseil-cpp/archive/5d8fc9192245f0ea67094af57399d7931d6bd53f.tar.gz"],
+  )
+
+def boost():
+  http_archive(
+    name = "boost",
+    build_file = "//bazel:BUILD.boost",
+    strip_prefix = "boost_1_70_0",
+    urls = [
+        "https://dl.bintray.com/boostorg/release/1.70.0/source/boost_1_70_0.tar.gz",
+        "https://downloads.sourceforge.net/project/boost/boost/1.70.0/boost_1_70_0.tar.gz",
+    ],
+    sha256 = "882b48708d211a5f48e60b0124cf5863c1534cd544ecd0664bb534a4b5d506e9",
+  )
+
+def com_github_redis_hiredis():
+  http_archive(
+    name = "com_github_redis_hiredis",
+    build_file = "//bazel:hiredis.BUILD",
+    strip_prefix = "hiredis-0.14.1",
+    urls = [
+      "https://github.com/redis/hiredis/archive/v0.14.1.tar.gz",
+    ],
+  )
+
+def com_github_sewenew_redis_plus_plus():
+  http_archive(
+    name = "com_github_sewenew_redis_plus_plus",
+    build_file = "//bazel:redis_plus_plus.BUILD",
+    strip_prefix = "redis-plus-plus-1.1.1",
+    urls = [
+      "https://github.com/sewenew/redis-plus-plus/archive/1.1.1.tar.gz",
+    ],
+  )
+
+def com_github_google_jwt_verify_lib():
+  http_archive(
+    name = "com_github_google_jwt_verify_lib",
+    sha256 = "7a5c35b7cbf633398503ae12cad8c2833e92b3a796eed68b6256d22d51ace5e1",
+    strip_prefix = "jwt_verify_lib-28efec2e4df1072db0ed03597591360ec9f80aac",
+    urls = [
+      "https://github.com/google/jwt_verify_lib/archive/28efec2e4df1072db0ed03597591360ec9f80aac.tar.gz"
+    ],
+  )
+
+def com_googlesource_boringssl():
+  http_archive(
+    name = "com_googlesource_boringssl",
+    sha256 = "15d855e5ec7c28b6b99159f1c6bbc7803e1623ed540f637174c6b88e7abd001c",
+    strip_prefix = "boringssl-936ca21922d266a31e3309144b082bdb3a689af7",
+    urls = [
+      "https://github.com/google/boringssl/archive/936ca21922d266a31e3309144b082bdb3a689af7.tar.gz"
+    ],
+  )
+
+def zlib():
+  http_archive(
+    name = "zlib",
+    build_file = "//bazel:zlib.BUILD",
```

Comment:
You should be able to use zlib BUILD file from protobuf so you don't need this file.

Reply:
`no such package '@zlib//': The repository '@zlib' could not be resolved and referenced by '//external:madler_zlib'` message had occurred on my environment.

Reply:
That's because you don't have `protobuf_deps()` called in WORKSPACE. https://github.com/protocolbuffers/protobuf/blob/master/protobuf_deps.bzl

Reply:
`protobuf_deps()` can be called on `grpc_extra_deps()` https://github.com/grpc/grpc/blob/v1.34.0/bazel/grpc_extra_deps.bzl#L29
And this is already called in WORKSPACE. Does it means we should download protobuf explicitly and calls `protobuf_deps()`?